apps/src/templates/codeReviewGroups/CodeReviewGroupsStatusToggle.jsx

```diff
@@ -0,0 +1,108 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import i18n from '@cdo/locale';
+import CodeReviewGroupsDataApi from './CodeReviewGroupsDataApi';
+import {setCodeReviewExpiresAt} from '../../redux/sectionDataRedux';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
+import ToggleSwitch from '@cdo/apps/code-studio/components/ToggleSwitch';
+import color from '@cdo/apps/util/color';
+
+function CodeReviewGroupsStatusToggle({
+  codeReviewExpiresAt,
+  sectionId,
+  setCodeReviewExpiration
+}) {
+  const [saveError, setSaveError] = useState(false);
+  const [saveInProgress, setSaveInProgress] = useState(false);
+  const currentTime = Date.now();
+  const isToggledOn = codeReviewExpiresAt
+    ? codeReviewExpiresAt > currentTime
+    : false;
+  // find days left by dividing milliseconds left by 1000 * 60 * 60 * 24. Round up so we
+  // do not show "in 0 days" on the last day.
+  const daysLeft =
+    isToggledOn &&
+    Math.ceil((codeReviewExpiresAt - currentTime) / (1000 * 60 * 60 * 24));
+  const api = new CodeReviewGroupsDataApi(sectionId);
+
+  const toggleEnableCodeReview = () => {
+    setSaveError(false);
+    setSaveInProgress(true);
+    const toggledValue = !isToggledOn;
+    api
+      .setCodeReviewEnabled(toggledValue)
+      .success(result => {
+        const newExpiration = result.expiration;
+        setCodeReviewExpiration(newExpiration);
+        setSaveInProgress(false);
+      })
+      .fail(() => {
+        setSaveError(true);
+        setSaveInProgress(false);
+      });
+  };
+
+  return (
+    <div>
+      <div style={styles.toggleAndError}>
+        <ToggleSwitch
+          isToggledOn={isToggledOn}
+          onToggle={toggleEnableCodeReview}
+          label={i18n.enableCodeReview()}
+        />
+        {saveInProgress && <Spinner style={styles.spinner} size="medium" />}
+        {saveError && (
+          <p style={styles.saveError}>
+            {isToggledOn
+              ? i18n.codeReviewToggleDisableError()
+              : i18n.codeReviewToggleEnableError()}
+          </p>
+        )}
+      </div>
+
+      {isToggledOn && (
+        <p style={styles.enabledMessage}>
+          {i18n.codeReviewAutoDisableMessage({daysLeft})}
+        </p>
+      )}
+    </div>
+  );
+}
+
+CodeReviewGroupsStatusToggle.propTypes = {
+  codeReviewExpiresAt: PropTypes.number,
+  sectionId: PropTypes.number,
+  setCodeReviewExpiration: PropTypes.func
+};
+
+export const UnconnectedCodeReviewGroupsStatusToggle = CodeReviewGroupsStatusToggle;
+
+export default connect(
```

Comment:
Not a big deal, but looks like we have a mix of redux and plumbing props for similar data (section data) in code review groups. For example - we have the sectionId in ManageCodeReviewGroups but are getting it via redux here. I don't think it's a huge deal either way but it might be cleaner for us to keep it consistent? Maybe ManageCodeReviewGroups obtains its props via redux (sectionId, codeReviewExpiresAt, setCodeReviewExpiration, etc) and passes those to its child components? Or we just use redux for all code review group components? I don't think it's blocking for this change, but curious what you think.

Reply:
I don't have a strong opinion here, I generally bias towards using state directly from redux if it already exists there. If we want to do something else though I am open to it. The reason I didn't pass it down from the parent here was because we only need the expiration date in this component. It felt weird to get it in a parent component and pass it down multiple levels, when we were only using it here.

Reply:
Yep makes sense. It might make sense to hook up CodeReviewGroupsManager to redux in that case (at some point later) but that's something we can also determine if/when we have more state across these components. Curious if @bencodeorg you had any thoughts here as well as you were working on the state and evaluating redux.

Reply:
I've been leaning towards not using Redux in this component, but I think that managing API calls introduces a bunch of state management that maybe I underestimated. Also, the work in this PR ties directly to modifying a Section object (which we already use redux for), whereas the other API call will only interact with code review groups/members (which has no existing redux). So might be reasonable to use Redux here and not elsewhere. I'm going to start the work on hooking up saving the groups themselves, so we'll see where that leads me :)